src/server/coprocessor/endpoint.rs

```diff
@@ -802,10 +651,10 @@ impl SelectContextCore {
         }
 
         self.topn_heap.as_mut().unwrap().try_add_row(h,
-                                                     self.order_cols.clone(),
-                                                     self.ctx.clone(),
                                                      values,
-                                                     sort_keys)
+                                                     sort_keys,
+                                                     self.order_cols.clone(),
+                                                     self.ctx.clone())
```

Comment:
I do not think change endpoint is a good idea in this PR.  Could we implement  `topn` based on current endpoint,  and do the  refactor for endpoint on another independent PR? @lishihai9017